Doc/tools/check-warnings.py

```diff
@@ -94,9 +235,10 @@ def fail_if_improved(
 def main() -> int:
     parser = argparse.ArgumentParser()
     parser.add_argument(
-        "--check-and-annotate",
-        help="Comma-separated list of files to check, "
-        "and annotate those with warnings on GitHub Actions",
+        "--check-and-annotate-changed",
+        nargs=2,
```

Comment:
> As a bonus, the script now works equally as well locally as remotely; just pass the commits/branches/tags to compare with `--check-and-annotate-changes <base> <branch>`, or use the default (comparing the changes on the current `HEAD` with the merge base in the local `main`). Handy!

Yes, sounds handy!

Although the defaults aren't usable via the CLI yet, or should I call it in another way?

```console
❯ python Doc/tools/check-warnings.py
❯ python Doc/tools/check-warnings.py --check-and-annotate-changed
usage: check-warnings.py [-h] [--check-and-annotate-changed CHECK_AND_ANNOTATE_CHANGED CHECK_AND_ANNOTATE_CHANGED] [--fail-if-regression] [--fail-if-improved]
check-warnings.py: error: argument --check-and-annotate-changed: expected 2 arguments
```

Reply:
Oops, I forgot to actually implement that. It ended up requiring a change to use `nargs="*"` and a manual length check, since it isn't possible to natively specify that an argument can take a specific variable number of values with argparse (see, e.g. [this SO answer](https://stackoverflow.com/questions/30944976/argparse-requiring-either-2-values-or-none-for-an-optional-argument)), but now it works, so you can pass any of 0, 1 or 2 arguments (just like `git diff`), but no more.